examples/generic_button_led/CMakeLists.txt

```diff
@@ -0,0 +1,7 @@
+# For more information about build system see
+# https://docs.espressif.com/projects/esp-idf/en/latest/api-guides/build-system.html
+# The following five lines of boilerplate have to be in your project's
+# CMakeLists in this exact order for cmake to work correctly
+cmake_minimum_required(VERSION 3.5)
```

Comment:
Please add `set(COMPONENTS main) # "Trim" the build. Include the minimal set of components; main and anything it depends on.` line

It will build only required components